### PR TITLE
2267 tcp scan event

### PR DESCRIPTION
**Pull request by ilija-lazoroski**
# What does this PR do?

Adds TCPScanEvent.
Issue #2267

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [x] Is the TravisCI build passing?
* [x] ~Was the CHANGELOG.md updated to reflect the changes?~
* [x] ~Was the documentation framework updated to reflect the changes?~
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by running unit tests
* [ ] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

File: vulture_allowlist.py [hunk 1]
```diff
@@ -29,6 +29,8 @@
 from monkey_island.cc.repository.zero_trust.IFindingRepository import IFindingRepository
 from monkey_island.cc.services import AgentSignalsService
 
+ge  # unused vairable (monkey/common/types.py:28)
+le  # unused variable (monkey/common/types.py:29)
```
Comment (mssalvatore) [inline vulture_allowlist.py:33]: Can we import NetworkPort and ignore `NetworkPort.ge` and `NetworkPort.le`?

File: monkey/common/types.py [hunk 2]
```diff
@@ -29,16 +41,16 @@ class PortStatus(Enum):
     An Enum representing the status of the port.
 
     This Enum represents the status of a network pork. The value of each
-    member is distincive and unique number.
+    member is the member's name in all lower-case characters.
     """
 
-    OPEN = 1
-    CLOSED = 2
+    OPEN = "open"
```
Comment (cakekoa) [inline monkey/common/types.py:47]: What prompted this change? Readability of serialized data?

File: monkey/common/types.py [hunk 3]
```diff
@@ -18,6 +18,18 @@
 MachineID: TypeAlias = PositiveInt
 
 
+class NetworkPort(ConstrainedInt):
```
Comment (cakekoa) [inline monkey/common/types.py:21]: Out of curiosity, does mypy play well with `ConstrainedInt`? I know `conint` and the like cause issues.

File: monkey/common/types.py (hunk 2 above)
Comment (mssalvatore) [reply monkey/common/types.py:47]: Yep, readability of serialized data.

File: monkey/common/types.py (hunk 3 above)
Comment (mssalvatore) [reply monkey/common/types.py:21]: I'm not seeing any compliants. See https://github.com/pydantic/pydantic/issues/239#issuecomment-409644889 for more details.

**Review by cakekoa — APPROVED** [review]

**Pull request merged**